src/test/daml/Daml/Finance/Instrument/Bond/Test/Util.daml

```diff
@@ -51,15 +54,46 @@ originateCallableBond depository issuer label description observers lastEventTim
       couponSchedule = createPaymentPeriodicSchedule firstCouponDate holidayCalendarIds
         businessDayConvention couponPeriod couponPeriodMultiplier issueDate maturityDate
       callSchedule = couponSchedule
+
+    -- Create a floating rate bond factory
+    callableBondFactoryCid <- toInterfaceContractId @Callable.F <$> submit issuer do
+      createCmd Callable.Factory with
+        provider = issuer
+        observers = M.empty
+
     -- CREATE_CALLABLE_BOND_INSTRUMENT_BEGIN
-    cid <- toInterfaceContractId <$> submitMulti [depository, issuer] [] do
-      createCmd Callable.Instrument with
-        depository; issuer; id = Id label; version = "0"; description
-        observers = M.fromList observers; lastEventTimestamp; prevEvents = []; couponSchedule
-        callSchedule; noticeDays; holidayCalendarIds; calendarDataProvider; dayCountConvention
-        useAdjustedDatesForDcf; floatingRate; couponRate; capRate; floorRate; currency; notional
+    let
+      instrumentId = Id label
+      instrumentVersion = "0"
+      bondInstrument = InstrumentKey with
+        issuer
+        depository
+        id = instrumentId
+        version = instrumentVersion
+
+    cid <- submitMulti [issuer] [calendarDataProvider] do
```

Comment:
Don't we need the authorisation of the depository in order to do this?

Also, given that this gets printed on the doc, would it be possible to grant the issuer visibility on the calendars, in order to avoid writing `submitMulti [issuer] [calendarDataProvider]`?

Reply:
publicParty was already an observer, so I used that as readAs instead.